### PR TITLE
feat(gateway): `lore doctor` and `lore setup status`

### LINKED ISSUE #869: Add `lore doctor` and `lore setup status` diagnostic commands
## Motivation

A new user ran `lore setup claude-code` before starting the gateway (and with a leftover Bedrock config present). Claude Code stopped connecting, and they had no way to see *what* lore had changed or *why* it was broken — they ended up manually unpicking environment variables to recover.

Today there is **no way to answer two basic questions**:

1. "What has lore configured, and where is each agent pointed?"
2. "Why won't my agent connect?"

Two commands close this gap. They also produce the config **inventory** that a future `lore setup undo`/`uninstall` command needs, so they should share one inventory module.

## `lore setup status`

Read-only inventory of what `lore setup` has touched. For each supported app, report the config file, whether it exists, and the current routing value:

- **Claude Code** — `~/.claude/settings.json` (`claudeCodeSettingsPath()`, `setup.ts:612`): `env.ANTHROPIC_BASE_URL`, `env.DISABLE_AUTO_COMPACT`, `env._CLAUDE_CODE_ASSUME_FIRST_PARTY_BASE_URL`.
- **Codex** — `~/.codex/config.toml` (`codexConfigPath()`, `setup.ts:256`): `openai_base_url`, `model_auto_compact_token_limit`.
- **OpenCode** — `~/.config/opencode/opencode.json` (`opencodeConfigPath()`, `setup.ts:476`): `provider.<id>.options.baseURL`, `compaction.auto`, and whether `@loreai/opencode` is in the `plugin` array.

For each: show whether the value points at a lore gateway, points elsewhere (and where), or is unset. Once provenance/backup lands, also show the saved original value.

## `lore doctor`

Broader health check that combines the inventory above with live + environment diagnostics:

- **Gateway reachability** — probe the configured URL(s) with `probeGateway()` (`start.ts:43`) and report up/down.
- **Port consistency** — compare the URL written by setup (defaults to `http://127.0.0.1:3207/v1`, `DEFAULT_PORT` at `setup.ts:215`) against the actually-running gateway's port via `readPortFile()` (`portfile.ts:46`). Flag the mismatch caused by the `3207 -> 5673 -> random` fallback chain (`start.ts:132`).
- **Conflicting shell env** — Claude Code's real environment variables override `settings.json` `env`. Detect when `ANTHROPIC_BASE_URL` / `OPENAI_BASE_URL` are exported in the shell and differ from what setup wrote (a major silent "it doesn't work and I can't tell why" source).
- **Bedrock/Vertex conflict** — detect `CLAUDE_CODE_USE_BEDROCK` / `CLAUDE_CODE_USE_VERTEX` / `ANTHROPIC_BEDROCK_BASE_URL`. Lore proxies the first-party Anthropic protocol and cannot proxy Bedrock/Vertex; pointing `ANTHROPIC_BASE_URL` at lore while these are set is exactly the broken state from the report. Warn clearly.
- **Plugin install** — for OpenCode, check `@loreai/opencode` is installed globally (mirror `isNpmPackageInstalled`, `setup.ts:109`) and registered.
- **Version** — print `lore` version.

Each finding should be a clear PASS / WARN / FAIL line with an actionable next step (e.g. "gateway not reachable -> run `lore start` (or `lore start --bg` once daemon mode lands), or undo with `lore setup undo claude-code`").

## Shared inventory module

Both commands (and the future `uninstall`) should read the same per-app inventory. Factor a single module that returns a structured description of each app's config file + current values, so status/doctor/undo never drift.

## Acceptance criteria

- [ ] `lore setup status` prints a per-app inventory (file path, exists?, current routing value, points-at-lore?).
- [ ] `lore doctor` runs the inventory plus: gateway reachability, port consistency, conflicting shell env, Bedrock/Vertex conflict, plugin install, version.
- [ ] Both are read-only (no config mutation).
- [ ] Findings use PASS/WARN/FAIL with actionable remediation.
- [ ] Shared inventory module consumed by status, doctor, and (later) uninstall.
- [ ] Help text + examples updated (`help.ts`).

## Related

Part of the broader setup-UX hardening from user feedback: prefer `lore run`, add `setup undo`/uninstall (inline/sidecar backups), active liveness check at setup time, daemon mode (`lore start --bg`), and Bedrock/Vertex detect-and-refuse (separate issue).

### PR DESCRIPTION
**Pull request by BYK**
## What

Two read-only diagnostic commands answering **"what has lore configured?"** and **"why won't my agent connect?"** — closes #869.

- **`lore setup status`** — prints a per-app inventory (file path, exists?, current routing value, points-at-lore?) for Claude Code, Codex, and OpenCode, plus whether a backup is present.
- **`lore doctor`** — runs the inventory plus:
  - Gateway reachability (`probeGateway` via `readPortFile`)
  - Port consistency (setup URL vs running gateway port — flags the `3207 → 5673 → random` fallback mismatch)
  - Conflicting shell-env overrides (`ANTHROPIC_BASE_URL` / `OPENAI_BASE_URL` exported in the shell differ from settings.json)
  - Bedrock/Vertex conflict (`CLAUDE_CODE_USE_BEDROCK` / `CLAUDE_CODE_USE_VERTEX` / `ANTHROPIC_BEDROCK_BASE_URL`)
  - OpenCode plugin install/registration consistency
  - Version
  - Each finding is PASS/WARN/FAIL with actionable remediation; exits non-zero on FAIL.

## Details

- New `cli/inventory.ts` — shared inventory module consumed by both commands (and the future uninstall):
  - `collectInventory()` → `AppInventory[]` (per-app file/rows/hasBackup)
  - `isLoreUrl()` / `classifyRoutingValue()` — routing classification (pure)
  - `runDoctorDiagnostics(input)` — pure findings generator (testable)
  - `commandDoctor()` — thin shell wiring real IO (port file, probe, env, npm)
- `lore setup status` dispatched in `commandSetup` (alongside `undo`).
- `lore doctor` dispatched in `main.ts` (`doctor` case + added to knownCommands).
- Help text + examples updated.

## Tests

- `inventory.test.ts` (12): `isLoreUrl`, `classifyRoutingValue`, `formatInventoryRow` edge cases.
- `doctor.test.ts` (9): `runDoctorDiagnostics` (gateway down/up, shell-env override, Bedrock conflict, port mismatch, plugin mismatch), `formatFinding`, integration `lore setup status` (missing + configured).
- Verified end-to-end via the CJS bundle in an isolated `HOME`: `lore doctor` on a clean box (FAIL: gateway down), after `lore setup claude-code` (inventory shows lore-routed URL + backup present), `lore setup status` prints the inventory.

## Verification

- Full suite: 3745 passed / 0 failed (32 skipped)
- Typecheck clean (all packages); lint exit 0
- Smoke-tested `lore doctor` and `lore setup status` via the CJS bundle

### CODE REVIEW

File: packages/gateway/src/cli/inventory.ts [hunk 1]
```diff
@@ -0,0 +1,464 @@
...
+  if (input.gatewayAlive && input.gatewayPort !== null) {
+    const expected = `127.0.0.1:${input.gatewayPort}`;
+    let mismatch = false;
+    for (const inv of input.inventory) {
+      for (const row of inv.rows) {
+        if (
+          row.routing.kind === "lore" &&
+          !row.routing.value.includes(expected)
+        ) {
+          mismatch = true;
```
[the remark this answers is not in the document]
Comment (BYK) [reply packages/gateway/src/cli/inventory.ts]: Fixed in e075b0fd. The port-consistency check now skips rows whose value doesn't start with `http` — the OpenCode plugin row (`@loreai/opencode`) is lore-routed but not a URL and was being incorrectly flagged. Added a regression test ("does NOT flag the OpenCode plugin row as a port mismatch").

File: packages/gateway/src/cli/inventory.ts [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+          label: "gateway reachable",
+          detail: "no gateway responding",
+          remediation:
+            "run `lore start --bg` (or `lore run` to launch an agent through the gateway)",
+        },
+  );
+
+  // 3. Port consistency: does any setup-routed URL match the running port?
+  if (input.gatewayAlive && input.gatewayPort !== null) {
+    const expected = `127.0.0.1:${input.gatewayPort}`;
```
[the remark this answers is not in the document]
Comment (BYK) [reply packages/gateway/src/cli/inventory.ts:329]: Fixed in ad312b22. Non-loopback URLs are now skipped — a remote gateway URL is intentional, not a mismatch against the local gateway. Added regression tests for both.

File: packages/gateway/src/cli/inventory.ts [hunk 3]
```diff
@@ -0,0 +1,472 @@
...
+        }
+      }
+    }
+    if (!mismatch) {
+      findings.push({
+        level: "PASS",
+        label: "port consistency",
+        detail: `all lore-routed URLs target port ${input.gatewayPort}`,
+      });
+    }
```
[the remark this answers is not in the document]
Comment (BYK) [reply packages/gateway/src/cli/inventory.ts]: Fixed in ad312b22. The PASS finding is only emitted when at least one URL was checked, so a non-URL-only inventory doesn't produce a misleading message. Added regression tests for both.

File: packages/gateway/src/cli/inventory.ts [hunk 4]
```diff
@@ -0,0 +1,484 @@
...
+        app: "Claude Code",
+        file,
+        fileExists,
+        key,
+        routing:
+          typeof v === "string" ? classifyRoutingValue(v) : { kind: "unset" },
+      });
+    }
+  }
+  return { app: "Claude Code", file, fileExists, rows, hasBackup };
```
[the remark this answers is not in the document]
Comment (BYK) [reply packages/gateway/src/cli/inventory.ts:156]: Fixed in 2a96519. collectClaudeCodeInventory now reads priorValue from the _loreBackup sidecar and passes it to the row for display.

File: packages/gateway/src/cli/inventory.ts [hunk 5]
```diff
@@ -0,0 +1,484 @@
...
+      CLAUDE_CODE_USE_BEDROCK: process.env.CLAUDE_CODE_USE_BEDROCK,
+      CLAUDE_CODE_USE_VERTEX: process.env.CLAUDE_CODE_USE_VERTEX,
+      ANTHROPIC_BEDROCK_BASE_URL: process.env.ANTHROPIC_BEDROCK_BASE_URL,
+    },
+    opencodePluginInstalled: isNpmPackageInstalledSafe("@loreai/opencode"),
+  });
+
+  // Print inventory first, then findings.
+  console.log("[lore] Setup inventory:");
+  printInventoryStatus();
```
[the remark this answers is not in the document]
Comment (BYK) [reply packages/gateway/src/cli/inventory.ts]: Fixed in 2a96519. commandDoctor now passes the already-collected inventory to printInventoryStatus — no double collect.
